typespec-tests/src/main/java/com/cadl/armresourceprovider/models/ManagedServiceIdentity.java

```diff
@@ -37,8 +35,7 @@ public final class ManagedServiceIdentity {
      * The identities assigned to this resource by the user.
      */
     @JsonProperty(value = "userAssignedIdentities")
-    @JsonInclude(value = JsonInclude.Include.NON_NULL, content = JsonInclude.Include.ALWAYS)
-    private Map<String, UserAssignedIdentity> userAssignedIdentities;
+    private UserAssignedIdentities userAssignedIdentities;
```

Comment:
This is a potential problem for ARM, as Swagger is a `Map`.
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/maps/azure-resourcemanager-maps/src/main/java/com/azure/resourcemanager/maps/models/ManagedServiceIdentity.java#L43-L45

We may reconsider whether `model Model is Record<>` to be a model class or just a Map.

However, I don't know why UserAssignedIdentity is not generated).

Reply:
For `ManagedServiceIdentity`, I believe no.

Reply:
Seems we have an extra `SchemaCleanup` logic to clean up unused schemas:
https://github.com/Azure/autorest.java/blob/ca8a1f1f9462c722cfaef60acfa0aa81a13ab1ad/fluentnamer/src/main/java/com/azure/autorest/fluent/transformer/FluentTransformer.java#L60

We have some logic to calculate which schemas to generate/not generate. Though here we have an `ObjectSchema`(UserAssignedIdentities) that has `DictionarySchema`(Record) as its parent, not as one of its properties, which we'll count as inUse(Line 82):
https://github.com/Azure/autorest.java/blob/ca8a1f1f9462c722cfaef60acfa0aa81a13ab1ad/fluentnamer/src/main/java/com/azure/autorest/fluent/transformer/SchemaCleanup.java#L73-L87

We did not count in Dictionary as parent case. Guess we could do either, add this particular case, or use TypeSpec's usage(do not use SchemaCleanup in TypeSpec)?

Reply:
I may slightly prefer "use usage info and do not use SchemaCleanup in TSP" as this is more straightforward, and in the future, it's easier to debug, but not sure if it will cause other issue in mgmt sdk.

Reply:
"We may reconsider whether model Model is Record<> to be a model class or just a Map. That can be decided later."

For this, we discussed before, the pure Record case, and previously our agreement is to generate the model to align with other language. Because later the model can add other properties, if we use `Map`, there will be breaking changes.

Reply:
> Currently we don't use TypeSpec usage in mpg... Instead, we have an extra `SchemaCleanup` logic to clean up unused schemas:
> 
> https://github.com/Azure/autorest.java/blob/ca8a1f1f9462c722cfaef60acfa0aa81a13ab1ad/fluentnamer/src/main/java/com/azure/autorest/fluent/transformer/FluentTransformer.java#L60
> 
> In fluent, we have specific logic to calculate which schemas to generate/not generate. Though here we have an `ObjectSchema`(UserAssignedIdentities) that has `DictionarySchema`(Record) as its parent, not as one of its properties, which we'll count as inUse(Line 82):
> 
> https://github.com/Azure/autorest.java/blob/ca8a1f1f9462c722cfaef60acfa0aa81a13ab1ad/fluentnamer/src/main/java/com/azure/autorest/fluent/transformer/SchemaCleanup.java#L73-L87
> 
> We did not count in Dictionary as parent case. Guess for mgmt TypeSpec, we should use TypeSpec's usage instead of SchemaCleanup.

This one seems a bug in that calculation. Let me know if you can fix it, or me to.

Reply:
@haolingdong-msft 
I don't think it worth a discussion... Seems just what service can/cannot do in client.tsp. There wasn't strict rule on this from the beginning.

Reply:
I can add this case in SchemaCleanup.